src/AppInstallerRepositoryCore/Rest/Schema/1_0/Interface.cpp

```diff
@@ -86,58 +93,61 @@ namespace AppInstaller::Repository::Rest::Schema::V1_0
 
         // TODO: Handle continuation token.
         HttpClientHelper clientHelper(GetSearchEndpoint(restApiUri));
-        json::value jsonObject = clientHelper.HandlePost(GetSearchBody(request));
+        web::json::value jsonObject = clientHelper.HandlePost(GetSearchBody(request));
 
         // Parse json and add results to SearchResult.
         if (jsonObject.is_null())
         {
             return result;
         }
 
-        auto& dataArray = jsonObject.at(Data).as_array();
+        auto& dataArray = jsonObject.at(GetJsonKeyNameString(Data)).as_array();
 
         for (auto& manifestItem : dataArray)
         {
-            std::string packageId = GetStringFromJsonStringValue(manifestItem.at(PackageIdentifier));
-            std::string packageName = GetStringFromJsonStringValue(manifestItem.at(PackageName));
-            std::string publisher = GetStringFromJsonStringValue(manifestItem.at(Publisher));
-            std::string packageFamilyName = GetStringFromJsonStringValue(manifestItem.at(PackageFamilyName));
-            std::string productCode = GetStringFromJsonStringValue(manifestItem.at(ProductCode));
-            json::value versionValue = manifestItem.at(Versions);
-
-            if (packageId.empty() || packageName.empty() || publisher.empty() || versionValue.is_null() || versionValue.as_array().size() == 0)
+            std::optional<std::string> packageId = GetStringFromJsonStringValue(manifestItem.at(GetJsonKeyNameString(PackageIdentifier)));
+            std::optional<std::string> packageName = GetStringFromJsonStringValue(manifestItem.at(GetJsonKeyNameString(PackageName)));
+            std::optional<std::string> publisher = GetStringFromJsonStringValue(manifestItem.at(GetJsonKeyNameString(Publisher)));
+            std::optional<std::string> packageFamilyName = GetStringFromJsonStringValue(manifestItem.at(GetJsonKeyNameString(PackageFamilyName)));
+            std::optional<std::string> productCode = GetStringFromJsonStringValue(manifestItem.at(GetJsonKeyNameString(ProductCode)));
+            web::json::value versionValue = manifestItem.at(GetJsonKeyNameString(Versions));
+
+            if (!packageId.has_value() || !packageName.has_value() || !publisher.has_value() || versionValue.is_null() || versionValue.as_array().size() == 0)
             {
                 continue;

```

Comment:
It seems like at least a verbose log line is in order to indicate an error in the incoming data.